src/cocotb/handle.py

```diff
@@ -64,18 +64,16 @@ def _value_limits(n_bits, limits):
 class SimHandleBase(ABC):
     """Base class for all simulation objects.
 
-    We maintain a handle which we can use for GPI calls.
+    All simulation objects are hashable and equatable by identity.
+
+    .. code-block:: python3
+        a = dut.clk
+        b = dut.clk
+        assert a == b
     """
 
     @abstractmethod
-    def __init__(self, handle, path):
-        """
-        .. Constructor. This RST comment works around sphinx-doc/sphinx#6885
-
-        Args:
-            handle (int): The GPI handle to the simulator object.
-            path (str): Path to this handle, ``None`` if root.
-        """
+    def __init__(self, handle: simulator.gpi_sim_hdl, path: str) -> None:
```

Comment:
`SimHandle` has a `None` default for **path**. We could do this change or make root an empty string instead.
```suggestion
    def __init__(self, handle: simulator.gpi_sim_hdl, path: Optional[str]) -> None:
```